pkg/features/patch_kube_features.go

```diff
@@ -0,0 +1,23 @@
+package features
+
+import (
+	"k8s.io/apimachinery/pkg/util/runtime"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+)
+
+var (
+	// owner: @fbertina
+	// beta: v1.23
+	//
+	// Enables the vSphere CSI migration for the Attach/Detach controller (ADC) only.
+	ADCCSIMigrationVSphere featuregate.Feature = "ADC_CSIMigrationVSphere"
```

Comment:
Instead of this, I think we want a 

```go
func OpenshiftStartCSMigrationVSphere() bool{
    if os.Getenv("OPENSHIFT_DO_VSPHERE_MIGRATION") == "YOLO!" {
        return true
    }
    return false.
}
```

And then we use that inline as `features.OpenshiftStartCSMigrationVSphere()`.

Reply:
We still need this ADC patch, but I added `OpenShiftStartCSIMigrationVSphere` in this file to contain the env check.